src/mixins/ReactView/ReactView.js

```diff
@@ -0,0 +1,60 @@
+import ReactDOM from 'react-dom'
+import Marionette from 'backbone.marionette'
+
+export const COMPONENT_UPDATED = 'component:updated'
+
+const ReactView = View =>
+  View.extend({
+    initialize(...args) {
+      if (!(this instanceof Marionette.ItemView)) {
+        console.warn('ReactView mixin is intended for use with ItemViews - behavior may not match expectations')
+      }
+      View.prototype.initialize.apply(this, args)
+      this.listenTo(this, 'close', () => this.teardownComponent())
+    },
+
+    render() {
+      this._ensureElement()
+      this._triggerBeforeRendered()
+      const props = this.serializeData()
+      const isRerender = !!this._component
+      this._component = this.template(props)
+      if (isRerender) {
+        // If we are re-rendering, we want to call ReactDOM.render() to refresh
+        this.renderComponent()
+      }
+      this._triggerRendered()
+    },
+
+    _triggerBeforeRendered() {
+      this.triggerMethod('before:render', this)
+      this.triggerMethod('item:before:render', this)
+    },
+
+    _triggerRendered() {
+      this.triggerMethod('render', this)
+      this.triggerMethod('item:rendered', this)
+    },
```

Comment:
I take it these are meant to be private methods. What do you think about moving them outside the view? Can define them like: 

```
const triggerRendered(view) {
  view.triggerMethod('render', view)
}
```

Reply:
Basically, yeah. I really kind of hate this way to make "private methods" lol. I get _why_, but passing the `view` instance in just kills me. I really don't care very much about protecting "private" methods in JS, I think I'd rather just make them `this.triggerRendered()` than do the passing-this-in-as-a-param thing. I just really hate that way of doing private methods in JS, idk what do you think?

Reply:
Was just a thought. I'm not too fussed about this as these methods seem pretty harmless and won't be documented. Fine with keeping it as is.